services/dynamic-sidecar/src/simcore_service_dynamic_sidecar/modules/attribute_monitor/_logging_event_handler.py

```diff
@@ -0,0 +1,226 @@
+    def __init__(
+        self,
+        path_to_observe: Path,
+        health_check_queue: AioQueue,
+        heart_beat_interval_s: PositiveFloat,
+    ) -> None:
+        # NOTE: runs in asyncio thread
+
+        self.path_to_observe: Path = path_to_observe
+        self.health_check_queue: AioQueue = health_check_queue
+        self.heart_beat_interval_s: PositiveFloat = heart_beat_interval_s
+
+        # This is accessible from the creating process and from
+        # the process itself and is used to stop the process.
+        self._stop_queue: AioQueue = aioprocessing.AioQueue()
+
+        self._file_system_event_handler: Optional[_LoggingEventHandler] = None
+        self._process: Optional[AioProcess] = None
+
+    def start_process(self) -> None:
+        # NOTE: runs in asyncio thread
+
+        with log_context(
+            logger,
+            logging.DEBUG,
+            f"{_LoggingEventHandlerProcess.__name__} start_process",
+        ):
+            self._process = aioprocessing.AioProcess(
+                target=self._process_worker, daemon=True
+            )
+            self._process.start()
+
+    def _stop_process(self) -> None:
+        # NOTE: runs in asyncio thread
+
+        with log_context(
+            logger,
+            logging.DEBUG,
+            f"{_LoggingEventHandlerProcess.__name__} stop_process",
+        ):
+            self._stop_queue.put(None)
+
+            if self._process:
+                # force stop the process
+                self._process.kill()
+                self._process.join()
+                self._process = None
+
+            # cleanup whatever remains
+            self._file_system_event_handler = None
+
+    def shutdown(self) -> None:
+        # NOTE: runs in asyncio thread
+
+        with log_context(
+            logger, logging.DEBUG, f"{_LoggingEventHandlerProcess.__name__} shutdown"
+        ):
+            self._stop_process()
+
+            # signal queue observers to finish
+            self.health_check_queue.put(None)
+
+    def _process_worker(self) -> None:
+        # NOTE: runs in the created process
+
+        observer = ExtendedInotifyObserver()
+        self._file_system_event_handler = _LoggingEventHandler()
+        watch = None
+
+        try:
+            watch = observer.schedule(
+                event_handler=self._file_system_event_handler,
+                path=f"{self.path_to_observe.absolute()}",
+                recursive=True,
+            )
+            observer.start()
+
+            while self._stop_queue.qsize() == 0:
+                # watchdog internally uses 1 sec interval to detect events
+                # sleeping for less is useless.
+                # If this value is bigger then the DEFAULT_OBSERVER_TIMEOUT
+                # the result will not be as expected. Keep sleep to 1 second
+
+                # NOTE: watchdog will block this thread for some period of
+                # time while handling inotify events
```

Comment:
hmm, maybe rework a comment of the function. but both comments one after the others are confusing.